src/lib/common/sol-certificate-impl-linux.c

```diff
@@ -43,36 +45,62 @@ static const char *const search_paths[] = {
     NULL,
 };
 
+static int
+get_home_config_dir(struct sol_buffer *buffer)
+{
+    int r;
+
+    r = sol_util_get_user_config_dir(buffer);
+    SOL_INT_CHECK(r, < 0, r);
+
+    return sol_buffer_append_printf(buffer, "/certs");
+}
+
+static inline bool
+is_cert(const char *file)
+{
+    struct stat st;
+
+    return stat(file, &st) == 0 && S_ISREG(st.st_mode) && st.st_mode & S_IRUSR;
```

Comment:
why checking mode is required? What if no user permission but group allows that? eventually it's better to use `access(file, R_OK)`?